src/CattyTests/DataModel/VariablesContainerTest.swift

```diff
@@ -236,4 +236,152 @@ final class VariablesContainerTest: XCTestCase {
+
+        var variables = container.objectVariables(for: objectA) as! [UserVariable]
+
+        XCTAssertEqual(1, variables.count)
+        XCTAssertEqual(variables[0].value as! Int, 20)
+
+        container.change(userVariable1, byValue: 10)
+
+        variables = container.objectVariables(for: objectA) as! [UserVariable]
+
+        XCTAssertEqual(1, variables.count)
+        XCTAssertEqual(variables[0].value as! Int, 30)
+    }
+
+    func testAddToUserList() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let list1 = UserVariable(name: "testVariable1", isList: true)
+
+        let container = VariablesContainer()
+
+        container.addObjectList(list1, for: objectA)
+        container.add(toUserList: list1, value: 10)
+        container.add(toUserList: list1, value: 20)
+
+        let lists = container.allLists() as! [UserVariable]
+        let itemInList = lists[0].value as! [Int]
+
+        XCTAssertEqual(2, itemInList.count)
+        XCTAssertEqual(itemInList[0], 10)
+        XCTAssertEqual(itemInList[1], 20)
+    }
+
+    func testDeleteFromUserList() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let list1 = UserVariable(name: "testVariable1", isList: true)
+
+        let container = VariablesContainer()
+
+        container.addObjectList(list1, for: objectA)
+        container.add(toUserList: list1, value: 10)
+        container.add(toUserList: list1, value: 20)
+
+        var lists = container.allLists() as! [UserVariable]
+        var itemInList = lists[0].value as! [Int]
+
+        XCTAssertEqual(2, itemInList.count)
+
+        container.delete(fromUserList: list1, atIndex: 2)
+
+        lists = container.allLists() as! [UserVariable]
+        itemInList = lists[0].value as! [Int]
+
+        XCTAssertEqual(1, itemInList.count)
+        XCTAssertEqual(itemInList[0], 10)
+    }
+
+    func testInsertToUserList() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let list1 = UserVariable(name: "testVariable1", isList: true)
+
+        let container = VariablesContainer()
+
+        container.addObjectList(list1, for: objectA)
+        container.insert(toUserList: list1, value: 10, atIndex: 1)
+        container.insert(toUserList: list1, value: 30, atIndex: 2)
+
+        var lists = container.allLists() as! [UserVariable]
+        var itemInList = lists[0].value as! [Int]
+
+        XCTAssertEqual(itemInList.count, 2)
+        XCTAssertEqual(itemInList[0], 10)
+        XCTAssertEqual(itemInList[1], 30)
+
+        container.insert(toUserList: list1, value: 20, atIndex: 2)
+
+        lists = container.allLists() as! [UserVariable]
+        itemInList = lists[0].value as! [Int]
+
+        XCTAssertEqual(itemInList.count, 3)
+        XCTAssertEqual(itemInList[0], 10)
+        XCTAssertEqual(itemInList[1], 20)
+        XCTAssertEqual(itemInList[2], 30)
+    }
+
+    func testReplaceItemInUserList() {
```

Comment:
Could you please also add another test method (i.e. `testReplaceItemInUserListinvalidIndex`) documenting what happens when an invalid index is passed to the method (e.g. which does not exist)?